tls/s2n_async_pkey.c

```diff
@@ -315,8 +314,7 @@ int s2n_async_pkey_op_apply(struct s2n_async_pkey_op *op, struct s2n_connection
      * protections in cases if caller frees connection object and then tries to resume
      * the connection. */
     POSIX_ENSURE(op->conn == conn, S2N_ERR_ASYNC_WRONG_CONNECTION);
-    POSIX_ENSURE(conn->handshake.async_state != S2N_ASYNC_INVOKING_CALLBACK, S2N_ERR_ASYNC_APPLY_WHILE_INVOKING);
-    POSIX_ENSURE(conn->handshake.async_state == S2N_ASYNC_INVOKED_WAITING, S2N_ERR_ASYNC_WRONG_CONNECTION);
+    POSIX_ENSURE(conn->handshake.async_state == S2N_ASYNC_INVOKED, S2N_ERR_ASYNC_WRONG_CONNECTION);
```

Comment:
Do we have a test for this error condition?

Reply:
It's not a new error condition, I just changed the name of the state. It doesn't look like there's a test.

Reply:
It's not a new error condition, I just changed the name of the state. Looks like there is a test:
https://github.com/aws/s2n-tls/blob/dffd2aa495549249018ee9334a4a4c661d49eb08/tests/unit/s2n_async_pkey_test.c#L72-L73